src/app/public/modules/country-field/country-field.component.ts

```diff
@@ -0,0 +1,295 @@
+  }
+
+  private control: AbstractControl;
+
+  private defaultCountryData: SkyCountryFieldCountry;
+
+  private idle: Subject<any> = new Subject();
+
+  private isFirstChange: boolean = true;
+
+  private _defaultCountry: string;
+
+  private _selectedCountry: SkyCountryFieldCountry;
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private formBuilder: FormBuilder,
+    private windowRef: SkyAppWindowRef
+  ) {
+    /**
+     * The json functions here ensures that we get a copy of the array and not the global original.
+     * This ensures that multiple instances of the component don't overwrite the original data.
+     *
+     * We must type the window object as any here as the intl-tel-input library adds its object
+     * to the main window object.
+     */
+    this.countries = JSON.parse(JSON.stringify((window as any)
+      .intlTelInputGlobals.getCountryData()));
+    this.defaultCountryData = this.countries.find(country => country.iso2 === 'us');
+
+    this.countrySearchForm = this.formBuilder.group({
+      countrySearch: new FormControl()
+    });
+
+    if ((<HTMLElement>elRef.nativeElement.parentElement)
+      .classList.contains('sky-phone-field-country-search')) {
+      this.isInPhoneField = true;
+    }
+  }
+
+  public ngOnInit(): void {
+    if (!this.defaultCountry) {
+      this.defaultCountry = 'us';
+    }
+
+    this.countrySearchForm.get('countrySearch').valueChanges.subscribe(newValue => {
+      if (newValue) {
+        this.selectedCountry = newValue;
+      }
+    });
+    this.addEventListeners();
+  }
+
+  public ngOnDestroy(): void {
+    this.selectedCountryChange.complete();
+    this.removeEventListeners();
+  }
+
+  /**
+   * Sets the country to validate against based on the county's iso2 code.
+   * @param countryCode The International Organization for Standardization's two-letter code
+   * for the default country.
+   */
+  public onCountrySelected(newCountry: SkyAutocompleteSelectionChange): void {
+    if (newCountry.selectedItem) {
+      this.writeValue(this.countries.find(countryInfo => countryInfo.iso2 ===
+        newCountry.selectedItem.iso2));
+    } else {
+      this.writeValue(undefined);
+    }
+  }
+
+  // Angular automatically constructs these methods.
+  /* istanbul ignore next */
+  public onChange: Function = (value: SkyCountryFieldCountry) => { };
+
+  /* istanbul ignore next */
+  public onTouched: Function = () => { };
+
+  public registerOnChange(fn: (value: any) => void): void {
+    this.onChange = fn;
+  }
+
+  public registerOnTouched(fn: () => void): void {
+    this.onTouched = fn;
+  }
+
+  // Allows Angular to disable the input.
+  public setDisabledState(disabled: boolean): void {
+    this.removeEventListeners();
+
+    if (!disabled) {
+      this.addEventListeners();
+    }
+
+    this.disabled = disabled;
+    this.changeDetector.markForCheck();
+  }
+
+  public validate(control: AbstractControl): ValidationErrors {
```

Comment:
When the control is set as `required`, and a user blurs out of the input with nothing inserted, the `ng-valid` class is still applied to the control. If you compare that with a simple input element, you'll see there should be an `ng-invalid` state applied instead. Do we need to check for validation on blur?

Reply:
I believe this is fixed. I think I was just missing the NG_VALIDATORS provider which I pushed as a part of my last commit. Let me know if you're still seeing weirdness but I put an input beside the country field and they seem to be working the same now.